core/src/main/java/org/apache/druid/math/expr/Expr.java

```diff
@@ -338,8 +336,8 @@ public BindingDetails withScalarArguments(Set<Expr> scalarArguments)
     {
       Set<IdentifierExpr> moreScalars = new HashSet<>();
       for (Expr expr : scalarArguments) {
-        final IdentifierExpr stringIdentifier = expr.getIdentifierExprIfIdentifierExpr();
-        if (stringIdentifier != null) {
+        final boolean isIdentiferExpr = expr.getIdentifierExprIfIdentifierExpr() != null;
+        if (isIdentiferExpr) {
```

Comment:
does this change really contribute to perf? if not then older version is more conventional IMHO.

Reply:
That change was to suppress an intellij inspection warning

Reply:
I still think original version was good enough, but I see that the reasoning as well. I think following could be best of both worlds. but all this is probably nit now.
```
if (expr.getIdentifierExprIfIdentifierExpr() != null) {
..
}
```

Reply:
@ccaominh this inspection is always a false-positive. I think it's worth disabling it in the Druid config.